tests/unit/models/ObjectLockConfiguration.js

```diff
@@ -0,0 +1,168 @@
+const assert = require('assert');
+const { parseString } = require('xml2js');
+
+const ObjectLockConfiguration =
+    require('../../../lib/models/ObjectLockConfiguration.js');
+
+function checkError(parsedXml, err, errMessage, cb) {
+    const config = new ObjectLockConfiguration(parsedXml).
+        getValidatedObjectLockConfiguration();
+    assert.strictEqual(config.error[err], true);
+    assert.strictEqual(config.error.description, errMessage);
+    cb();
+}
+
+function generateRule(testParams) {
+    if (testParams.key === 'Rule') {
+        return `<Rule>${testParams.value}</Rule>`;
+    }
+    if (testParams.key === 'DefaultRetention') {
+        return `<Rule><DefaultRetention>${testParams.value} ` +
+            '</DefaultRetention></Rule>';
+    }
+    const mode = testParams.key === 'Mode' ?
+        `<Mode>${testParams.value}</Mode>` : '<Mode>GOVERNANCE</Mode>';
+
+    let time = '<Days>1</Days>';
+    if (testParams.key === 'Days') {
+        time = `<Days>${testParams.value}</Days>`;
+    }
+    if (testParams.key === 'Years') {
+        time = `<Years>${testParams.value}</Years>`;
+    }
+    return `<Rule><DefaultRetention>${mode}${time}</DefaultRetention></Rule>`;
+}
+
+function generateXml(testParams) {
+    const Enabled = testParams.key === 'ObjectLockEnabled' ?
+        `<ObjectLockEnabled>${testParams.value}</ObjectLockEnabled>` :
+        '<ObjectLockEnabled>Enabled</ObjectLockEnabled>';
+    const Rule = generateRule(testParams);
+    const ObjectLock = testParams.key === 'ObjectLockConfiguration' ? '' :
```

Comment:
I'm confused by this line, what if the key was `EmptyObjectLockConfiguration` for example?

Reply:
it's just following the same pattern for setting up the object lock config to test like the rest of the keys. The only difference for `ObjectLockConfiguration` is that its only error condition is if it's empty